src/Raven.Server/Documents/Sharding/Background/ShardedDocumentsMigrator.cs

```diff
@@ -1,38 +1,37 @@
 ﻿using System;
+using System.Threading;
 using System.Threading.Tasks;
-using Raven.Server.Background;
 using Raven.Server.ServerWide.Commands.Sharding;
 using Raven.Server.ServerWide.Context;
 using Sparrow.Logging;
 
 namespace Raven.Server.Documents.Sharding.Background
 {
-    public class ShardedPeriodicDocumentsMigrator : BackgroundWorkBase
+    public class ShardedDocumentsMigrator : IDisposable
     {
         private readonly ShardedDocumentDatabase _database;
+        private readonly CancellationTokenSource _cts;
+        private readonly Logger _logger;
 
-        public ShardedPeriodicDocumentsMigrator(ShardedDocumentDatabase database) : base(database.ShardedDatabaseName, database.ShardedDocumentsStorage.DocumentDatabase.DatabaseShutdown)
+        public ShardedDocumentsMigrator(ShardedDocumentDatabase database)
         {
             _database = database;
+            _cts = CancellationTokenSource.CreateLinkedTokenSource(database.ShardedDocumentsStorage.DocumentDatabase.DatabaseShutdown);
+            _logger = LoggingSource.Instance.GetLogger(database.Name, GetType().FullName);
         }
 
-        protected override async Task DoWork()
+        internal async Task<bool> ExecuteMoveDocumentsAsync()
         {
-            await WaitOrThrowOperationCanceled(_database.Configuration.Sharding.PeriodicDocumentsMigrationInterval.AsTimeSpan);
+            _cts.Token.ThrowIfCancellationRequested();
 
             if (_database.ServerStore.Sharding.HasActiveMigrations(_database.ShardedDatabaseName))
-                return;
+                return false;
 
-            await ExecuteMoveDocumentsAsync();
-        }
-
-        internal async ValueTask ExecuteMoveDocumentsAsync()
-        {
+            bool found = false;
             try
             {
                 int bucket = -1;
                 int moveToShard = -1;
-                bool found = false;
```

Comment:
What happens if we close the database when this is running?

Reply:
done